flutter_local_notifications/lib/src/platform_flutter_local_notifications.dart

```diff
@@ -54,6 +54,15 @@ class MethodChannelFlutterLocalNotificationsPlugin
 /// Android implementation of the local notifications plugin
 class AndroidFlutterLocalNotificationsPlugin
     extends MethodChannelFlutterLocalNotificationsPlugin {
+  /// Android specific plugin instance or null when not running in Android environment
+  static AndroidFlutterLocalNotificationsPlugin get instance {
+    if (FlutterLocalNotificationsPlatform.instance
```

Comment:
This code block, including the if statement could be replaced by casting using the `as` keyword that was used in the earlier version of your PR. That is, it'll look like `FlutterLocalNotificationsPlatform.instance as AndroidFlutterLocalNotificationsPlugin`

If the type doesn't match it will return null as well :)

Reply:
oh, didn't know that `as` will return you `null`, was assuming that this would result in something like `ClassCastException` (yeah, can't hide my Java background ;))

Reply:
Actually you might need to double check in case I'm wrong. Perhaps it doesn't work the same way as C#

Reply:
Ok just tried it works fine :)
Reason I mentioned checking was I had used it when I knew the type was what I expected but can't remember if I had done this for when the type might not match as well. Perhaps the Dart docs around the [`as`](https://dart.dev/guides/language/language-tour#type-test-operators) keyword need updating hmm

Reply:
Hmm this question would suggest it doesn't work the same way https://stackoverflow.com/questions/52494535/type-conversion-using-as-in-dart

Let's keep what you originally had then. Apologies for the confusion and sorry about that

Reply:
Better be safe than sorry ;) I've reverted previous refactroing

Reply:
well, at least dartpat says that not checking types could backfire:
```
Uncaught Error: CastError: "asdf": type 'JSString' is not a subtype of type 'int'
```

Reply:
Cool, looks like I learned something, I guess it implies it works for subtypes but better to be cautious as you said :)